app/javascript/utils/debounce.js

```diff
@@ -0,0 +1,18 @@
+window._ = {
+  debounce: function(func, wait, immediate) {
+    var timeout
+
+    return function() {
+      var context = this, args = arguments
+
+      var later = function() {
+        timeout = null
+        if (!immediate) func.apply(context, args)
+      }
+      var callNow = immediate && !timeout
+      clearTimeout(timeout)
+      timeout = setTimeout(later, wait)
+      if (callNow) func.apply(context, args)
+    }
+  }
+}
```

Comment:
Here's what I would do:

```suggestion
export default function(func, wait, immediate) {
    var timeout

    return function() {
      var context = this, args = arguments

      var later = function() {
        timeout = null
        if (!immediate) func.apply(context, args)
      }
      var callNow = immediate && !timeout
      clearTimeout(timeout)
      timeout = setTimeout(later, wait)
      if (callNow) func.apply(context, args)
    }
  }
}
```

And then when you go to use it:

```javascript
# javascript/controllers/some_example.js
import debounce from "../utils/debounce.js"
import { Controller } from '@hotwired/stimulus'

export default Class extends Controller {
 // blah
}
```

-------

If you're set on it being a "collection" of utilities, I'd probably do this instead. The only material difference is exporting more than one thing from the module.

```javascript
export const debounce = function (...) {...}
```
and
```javascript
import { debounce } from "../utils"
```